ktlint/build.gradle.kts

```diff
@@ -76,11 +76,10 @@ tasks.register<Checksum>("shadowJarExecutableChecksum") {
     description = "Generates MD5 checksum for ktlint executable"
     group = "Distribution"
 
-    files = shadowJarExecutable.get().outputs.files
+    inputFiles.setFrom(shadowJarExecutable.get().outputs.files)
```

Comment:
While you're fixing deprecations, consider going fully lazy:
`inputFiles.setFrom(shadowJarExecutable.map { it.outputs.files })`
(Same for few line down.)

---

Below are some feedback on these tasks, not in scope in this PR just wanted to mention them.

---

Note that finalizedBy will execute checksum even on failure.

---

The recommendation from Gradle is to always create a full custom task, that way you don't need mapping like here, but you can just say `from(otherTask.outputDir)` where `outputDir` is a `DirectoryProvider`. Also caching and path sensitivity is not defined in register.

---

After writing all this, I realized that there's an overlapping output directory between tasks here. This is dangerous as noted by Gradle. The only example is see though is `cleanShadowJarExecutableChecksum` will delete the output of shadowJar too.

Reply:
a0737b4cb38eabc566d1fbf8162860dddfece391